sklearn/datasets/tests/test_samples_generator.py

```diff
@@ -146,6 +146,46 @@ def test_make_classification_informative_features():
              n_clusters_per_class=2)
 
 
+@pytest.mark.parametrize(
+    'weights, err_type, err_msg',
+    [
+        ([], ValueError,
+         "Weights specified but incompatible with number of classes."),
+        ([.25, .75, .1], ValueError,
+         "Weights specified but incompatible with number of classes."),
+        (np.array([]), ValueError,
+         "Weights specified but incompatible with number of classes."),
+        (np.array([.25, .75, .1]), ValueError,
+         "Weights specified but incompatible with number of classes."),
+        (np.random.random(3), ValueError,
+         "Weights specified but incompatible with number of classes.")
+    ]
+)
+def test_make_classification_weights_type(weights, err_type, err_msg):
+    with pytest.raises(err_type, match=err_msg):
+        make_classification(weights=weights)
+
+
+def test_make_classification_weights_array_or_list_ok():
+    X1, y1 = make_classification(weights=[.1, .9],
+                                 random_state=0)
+    X2, y2 = make_classification(weights=np.array([.1, .9]),
+                                 random_state=0)
+    assert (X1.all() == X2.all()) and (y1.all() == y2.all())
```

Comment:
`X1.all()` returns True if X1 is all non-zero. Is this assertion to do the following:

```py
assert_almost_equal(X1, X2)
assert_almost_equal(y1, y2)
```